java-frontend/src/main/java/org/sonar/plugins/java/api/semantic/Symbol.java

```diff
@@ -174,6 +174,12 @@
      */
     List<Type> parameterTypes();
 
+    /**
+     * Symbols of parameters declared by this method.
+     * @return empty list if methods has not parameters
+     */
+    List<Symbol> declarationParameters();
```

Comment:
We should add in the javadoc of this API:

- For `List<Type> parameterTypes();`: in case of generic types of method invokations, this list of types is more specialized than `declarationParameters().stream().map(Symbol::type)`, for exemple it could be `String` instead of `T`.
- For `List<Symbol> declarationParameters();` useful to access parameter annotations but the drawback is: 1) in case of generic types of method invokations, this list of types is less specialized than `parameterTypes()`,  for exemple it could be `T` instead of `String`  2) when the declaration come from binaries, the name of the symbol will be generated `arg...`